components/ws-manager-mk2/controllers/status.go

```diff
@@ -40,7 +40,7 @@ func updateWorkspaceStatus(ctx context.Context, workspace *workspacev1.Workspace
 			workspace.Status.Phase = workspacev1.WorkspacePhasePending
 		}
 
-		if workspace.Status.Phase != workspacev1.WorkspacePhasePending {
+		if isDisposalFinished(workspace) {
```

Comment:
Bug found by the tests: it's possible for a workspace to get stopped and its pod to get deleted while the Phase is still `Pending`. It would then get stuck here and never transition to `Stopped`, causing the workspace resource to stay behind.

Instead, it now checks whether the workspace has been disposed (backup taken, content init failed, ...) and transitions to `Stopped` if so